llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1250,6 +1268,158 @@ void AddressSanitizerPass::printPipeline(
   OS << '>';
 }
 
+// Append a new argument "launch_data" to user's spir_kernel & spir_func
+static void ExtendSpirKernelArgs(Module &M, FunctionAnalysisManager &FAM) {
+  SmallVector<Function *> SpirFixupFuncs;
+  for (Function &F : M) {
```

Comment:
do we need to patch functions that doesn't use local_accessor?

Reply:
yes, launch_data are also used to store sanitizer report.